src/t_zset.c

```diff
@@ -2768,22 +2758,15 @@ void zunionInterDiffGenericCommand(client *c, robj *dstkey, int numkeysIndex, in
             zuiClearIterator(&src[i]);
         }
 
-        /* Step 2: convert the dictionary into the final sorted set. */
-        di = dictGetIterator(accumulator);
-
-        /* We now are aware of the final size of the resulting sorted set,
-         * let's resize the dictionary embedded inside the sorted set to the
-         * right size, in order to save rehashing time. */
-        dictExpand(dstzset->dict, dictSize(accumulator));
+        di = dictGetIterator(dstzset->dict);
 
         while ((de = dictNext(di)) != NULL) {
             sds ele = dictGetKey(de);
             score = dictGetDoubleVal(de);
             znode = zslInsert(dstzset->zsl, score, ele);
-            dictAdd(dstzset->dict, ele, &znode->score);
+            dictSetVal(dstzset->dict, de, &znode->score);
```

Comment:
I think we should leave the comment (`Step 2`) suggesting that we need to change the value to be in the format of sorted set.

Reply:
And in terms of safety, I think we are fine modify the entry value, if we were adding new entries to the dictionary I would have been a bit more skeptical.

Reply:
Yeah, you're right, that comment has been restored.